### PR TITLE
Fix wait-for-deployment

### PR DESCRIPTION
**Pull request by sjmiller609**
with regard to https://github.com/astronomer/issues/issues/917

TLDR boolean type casting an empty array is True in javascript. The int value 0 is False however.

I already committed this change directly to the release branch.

### CODE REVIEW

File: src/resolvers/subscription/deployment-status/index.js [hunk 1]
```diff
@@ -51,7 +51,9 @@ export async function subscribe(parent, args, { pubsub }) {
     const res = await Promise.resolve(getMetric(releaseName));
     publish({
       deploymentStatus: {
-        result: res && res.result ? res.result[0].value[1] : 0
+        result: res &&
+                res.result &&
+                res.result.length ? res.result[0].value[1] : 0
```
Comment (schnie) [inline src/resolvers/subscription/deployment-status/index.js]: Should this be `res.result.length > 0`?

Comment (sjmiller609) [reply src/resolvers/subscription/deployment-status/index.js]: no, that is redundant.

**Comment by sjmiller609** [comment]
not sure about the merge conflict I just tried to resolve

**Comment by sjmiller609** [comment]
ok looks good now

**Comment by sjmiller609** [comment]
got a lint error, fixing.

**Comment by sjmiller609** [comment]
I switched up the CI to run lint and unit testing without waiting for build first. but I did not put the build downstream of testing, because testing is sufficiently long enough for us to benefit from running them in parallel. Also, it's probably better to know security issues sooner rather than later.

**Comment by andriisoldatenko** [comment]
@ashb @schnie PTAL

File: package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "test": "prisma generate && jest",
     "coverage": "prisma generate && jest --coverage",
     "lint": "prisma generate && eslint src --ignore-pattern src/lib/generated --ext .js --ext .graphql",
+    "fix-lint": "prisma generate && eslint src --ignore-pattern src/lib/generated --ext .js --ext .graphql --fix",
```
Comment (andriisoldatenko) [inline package.json:12]: i think you can remove it:
`npm run lint -- --fix` should work.

Comment (sjmiller609) [reply package.json]: ty!

**Pull request merged**